micro-rdk/src/common/conn/viam.rs

```diff
@@ -442,10 +530,30 @@ where
         }
 
         let exec = self.executor.clone();
-        exec.block_on(Box::pin(self.run()));
+        let run_result = exec.block_on(Box::pin(self.run()));
+        match run_result.app_client_tasks_result {
+            Ok(()) => {}
+            Err(errors::ServerError::ServerTaskShutdownTimeout) => {
+                log::info!("app client tasks cancelled from timeout")
+            }
+            Err(err) => {
+                log::error!("app client tasks returned with error: {:?}", err);
+            }
+        }
+        match run_result.server_task_result {
+            Ok(()) => {}
+            Err(errors::ServerError::ServerTaskShutdownTimeout) => {
+                log::info!("server task cancelled from timeout")
+            }
+            Err(err) => {
+                log::error!("server task returned with error: {:?}", err);
+            }
+        }
+        exec.block_on(force_shutdown(run_result.app_client.clone()));
+        unreachable!()
```

Comment:
nit: so this doesn't get us back to main?

Reply:
Not sure how I would do that besides putting the shutdown call in `main.rs` (which doesn't seem like a good idea?)